File: 2313.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.split('\n');
const numbers = lines.shift().split(" ");
numbers.sort((a,b) => a - b).reverse();

const [a, b, c] = numbers
const A = parseFloat(a);
const B = parseFloat(b);
const C = parseFloat(c);

if(A >= B + C){
  console.log("Invalido");
  return;
}

if(A == B && B == C){
  console.log("Valido-Equilatero");
}
else if(A == B || A == C || B == C){
  console.log("Valido-Isoceles");
}

else if(A != B && B != C && C != A){
  console.log("Valido-Escaleno");
}

if(Math.pow(A, 2) == (Math.pow(B,2) + Math.pow(C,2))){
  console.log("Retangulo: S");
}
else{
  console.log("Retangulo: N");
}

File: java/2313.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.PrintWriter;
import java.util.Arrays;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter out = new PrintWriter(System.out);
        
        String[] numbersStr = br.readLine().trim().split(" ");
        double[] numbers = new double[3];
        
        for (int i = 0; i < 3; i++) {
            numbers[i] = Double.parseDouble(numbersStr[i]);
        }
        
      
        Arrays.sort(numbers);
        double A = numbers[2];
        double B = numbers[1];
        double C = numbers[0];
        
       
        if (A >= B + C) {
            out.println("Invalido");
            out.close();
            br.close();
            return;
        }
        
        if (A == B && B == C) {
            out.println("Valido-Equilatero");
        } else if (A == B || A == C || B == C) {
            out.println("Valido-Isoceles");
        } else {
            out.println("Valido-Escaleno");
        }
        
        
        if (Math.pow(A, 2) == (Math.pow(B, 2) + Math.pow(C, 2))) {
            out.println("Retangulo: S");
        } else {
            out.println("Retangulo: N");
        }
        
        out.close();
        br.close();
    }
}
